earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -18,19 +18,131 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 #include <time.h>
-
+#include <cstdlib>
 #include "common/performancelogger.h"
 #include "common/notify.h"
 #include "common/timeutils.h"
+#include <cassert>
 
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+string ioFileName;
+string threadFileName;
+string timeFileName;
+
+void generateFileNames() {
```

Comment:
This looks like it may be better done in a logger constructor before any of the file names get used.

Reply:
Would we want to have the different files to have the same timestamp or does it matter?

Reply:
Yeah, I think that's a good idea.  I don't think that would change if this function was a member function called from a constructor.  It would just insure these variables are initialized before logger methods are called.